tests/keithley_2700.py

```diff
@@ -0,0 +1,168 @@
+import unittest
+from utils.channel_access import ChannelAccess
+from utils.ioc_launcher import get_default_ioc_dir
+from utils.test_modes import TestModes
+from utils.testing import get_running_lewis_and_ioc, skip_if_recsim
+import time
+
+DEVICE_PREFIX = "KHLY2700_01"
+
+IOCS = [
+    {
+        "name": DEVICE_PREFIX,
+        "directory": get_default_ioc_dir("KHLY2700"),
+        "macros": {},
+        "emulator": "keithley_2700",
+    },
+]
+
+TEST_MODES = [TestModes.RECSIM, TestModes.DEVSIM]
+
+
+class Status(object):
+    ON = "ON"
+    OFF = "OFF"
+
+
+class Keithley_2700Tests(unittest.TestCase):
+    """
+    Tests for the Keithley2700.
+    """
+
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc("keithley_2700", DEVICE_PREFIX)
+        self.ca = ChannelAccess(default_timeout=30, device_prefix=DEVICE_PREFIX)
+        self.ca.wait_for("IDN")
+
+    def test_GIVEN_scan_state_set_WHEN_read_THEN_scan_state_is_as_expected(self):
+        sample_data = [0, 1]
+        expected_state = ["INT", "NONE"]
+        for set_value, expected_value in zip(sample_data, expected_state):
```

Comment:
I'm not a fan of zipping. It means you have to be certain your data structures are in the right order. If the test is ever modified, there's potential for confusion. If your data is logically connected, it should be obvious (if possible) at point of definition:

```
sample_data = {0: "INIT", 1: "NONE"}
for set_value, expected_value in sample_data.items():
    ...
```

Please apply throughout

Reply:
Removed zips